matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -119,12 +119,15 @@ public MapMatching(GraphHopper graphHopper, PMap hints) {
         RoutingAlgorithmFactory routingAlgorithmFactory = graphHopper.getAlgorithmFactory(profile.getName(), disableCH, disableLM);
         if (routingAlgorithmFactory instanceof CHRoutingAlgorithmFactory) {
             ch = true;
-            routingGraph = graphHopper.getGraphHopperStorage().getCHGraph(((CHRoutingAlgorithmFactory) routingAlgorithmFactory).getCHProfile());
+            CHProfile chProfile = ((CHRoutingAlgorithmFactory) routingAlgorithmFactory).getCHProfile();
+            routingGraph = graphHopper.getGraphHopperStorage().getCHGraph(chProfile);
+            weighting = chProfile.getWeighting();
         } else {
             ch = false;
             routingGraph = graphHopper.getGraphHopperStorage();
+            // since map matching does not support turn costs we have to disable them here explicitly
+            weighting = graphHopper.createWeighting(profile, hints, true);
```

Comment:
Here its important to disable turn costs since the map matching algorithm is running node-based